### DISCUSSION TITLE
Can't use matlab_ls

### DISCUSSION THREAD

**Discussion by NonadonSon** [LazyVim/LazyVim#1409] [Q&A]
I want to edit code of matlab by neovim. I have installed matlab-language-server by Mason, and here is my configuration:
```lua
return {
  "neovim/nvim-lspconfig",

  opts = {
    servers = {
      -- Ensure mason installs the server
      clangd = {
        keys = {
          { "<leader>cR", "<cmd>ClangdSwitchSourceHeader<cr>", desc = "Switch Source/Header (C/C++)" },
        },
        root_dir = function(fname)
          return require("lspconfig.util").find_git_ancestor(fname)
            or require("lspconfig.util").root_pattern("compile_commands.json")(fname)
            or require("lspconfig.util").root_pattern("Makefile")(fname)
            or require("lspconfig.util").root_pattern("xmake.lua")(fname)
        end,
        capabilities = {
          offsetEncoding = { "utf-16" },
        },
        cmd = {
          "clangd",
          "--background-index",
          "--clang-tidy",
          "--header-insertion=iwyu",
          "--completion-style=detailed",
          "--function-arg-placeholders",
          "--fallback-style=llvm",
          "--log=verbose",
          "--all-scopes-completion",
          "--query-driver=/usr/bin/g++-*",
        },
        init_options = {
          usePlaceholders = true,
          completeUnimported = true,
          clangdFileStatus = true,
        },
      },
      matlab_ls = {
        cmd = { "matlab-language-server", "--stdio" },
        matlab = {
          indexWorkspace = true,
          installPath = "/opt/matlab2021b/",
        },
        root_dir = function(fname)
          return require("lspconfig.util").find_git_ancestor(fname)
            or require("lspconfig.util").root_pattern("compile_commands.json")(fname)
            or require("lspconfig.util").root_pattern("Makefile")(fname)
            or require("lspconfig.util").root_pattern("xmake.lua")(fname)
        end,

        single_file_support = true,
      },
    },
  },
}
```
I open a m file but find matlab_ls can't launch. I try use command `LspInfo` to check reason and the following picture was obtained:
![image](https://github.com/LazyVim/LazyVim/assets/83964113/c741830f-d1a1-4dff-91ea-dbd81b4b1130)

**Comment by dpetka2001** [comment]
Not sure how much help my opinion can be. `cmd` is the same as default used from `nvim-lspconfig`, why redefine it? It also should be 
```lua
settings = {
	matlab = {
          indexWorkspace = true,
          installPath = "/opt/matlab2021b/",
    },
},
```
Other than that I would guess it would have to do something with the `root_dir` detection. You could also check `:LspLog` for possible errors that might help you more.

**Reply by NonadonSon** [reply]
Thank you. I use your method, but the situation has not changed, I still can't use matlab_ls.

**Comment by abeldekat** [comment]
Hello @ZZT-console,

Could you try from scratch, using stock [lazyvim starter](https://github.com/LazyVim/starter)?
I only added the following:

`lua/plugins/lsp.lua`:
```
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        matlab_ls = {},
        clangd = {},
      },
    },
  },
}
```

I cloned https://github.com/garciparedes/matlab-examples.git, and opened a matlab file.
The lsp attaches. 

Then, I copied your `clangd` and `matlab` config into `lua/plugins/lsp.lua`. Still, the lsp attaches.

Your screenshot shows the config of `matlab`. I don't know if its meaningful, but my LspInfo does not contain the lines "cmd is executable" and "custom handlers".

**Reply by dpetka2001** [reply]
I tried it also with fresh starter Lazyvim template and it does seem to attach.I just configured `settings` and `root_dir` according to OP's configuration (since `cmd` is the same as the default so I left it out) and it did attach to the file both when testing with a file inside a git repo and with a single file not in a git repo. In both cases the LSP server attached. However I'm unable to do any more testing since I don't own a Matlab licence and can't test if the server can connect to the Matlab instance installed on your computer.
